etcdserver/server.go

```diff
@@ -1039,7 +1047,12 @@ func (s *EtcdServer) stopWithDelay(d time.Duration, err error) {
 
 // StopNotify returns a channel that receives a empty struct
 // when the server is stopped.
-func (s *EtcdServer) StopNotify() <-chan struct{} { return s.done }
+func (s *EtcdServer) StopNotify() <-chan struct{} {
```

Comment:
we probably should also comment that stopNotify should only be called after start returns. or stopNotify can be pending forever. if we follow that convention, will race still happen?

Reply:
> stopNotify should only be called after start returns.

Now that `servePeers` start first, `v3rpc.newStreamInterceptor` with `v3rpc.monitorLeader` gets called before `e.Server.Start`, so there was race waiting on `done` channel.

> stopNotify can be pending forever if it gets the old done chan

it will timeout after election timeout

https://github.com/coreos/etcd/blob/15bfc1b36162805e8a90ae747d955667870c2f95/etcdserver/api/v3rpc/interceptor.go#L110-L118

> will race still happen

I think we still need locks because it will always be called before `e.Server.Start`?

Reply:
> Now that servePeers start first,

Why do we need this change?